python/app/graphql/resolvers/order_resolver.py

```diff
@@ -115,6 +116,12 @@ def updateOrderStatus(self, input: UpdateOrderStatusInput) -> Optional[Order]:
             if input.status not in OrderStatus.__members__:
                 raise ValueError(f"Invalid order status: {input.status}. Allowed: {list(OrderStatus.__members__.keys())}")
 
+            # ✅ If order status changes from READY_FOR_DELIVERY → another status, delete the assigned delivery record
+            delivery = db.query(DeliveryModel).filter(DeliveryModel.orderId == input.orderId).first()
+            if delivery and input.status != "READY_FOR_DELIVERY":
+                db.delete(delivery)
```

Comment:
What are we deleting here?
Also read about soft-delete. We should not delete any record